### PR TITLE
texture for trees

### PR DESCRIPTION
**Pull request by DJNGattuso**

### CODE REVIEW

File: NeverGreenGrove/NeverGreenGrove/src/NeverGreenGrove.cpp [hunk 1]
```diff
@@ -89,7 +89,7 @@ int main()
       tree2->Draw();
       tree3->Draw();
 
-	  glBindTexture(GL_TEXTURE_2D, terrain.getTexture());
+      glBindTexture(GL_TEXTURE_2D, terrain.getTexture());
```
Comment (prince-chrismc) [inline NeverGreenGrove/NeverGreenGrove/src/NeverGreenGrove.cpp]: this should be within the `TerrainChunk::Draw(...)`

File: NeverGreenGrove/NeverGreenGrove/src/Texture.h [hunk 2]
```diff
@@ -1,31 +1,51 @@
...
 
 class Texture
 {
 public:
-	Texture() {}
-	void createTexture(const char* imagePath);
-	float getCoordinates();
-	unsigned int getTexture();
+   Texture() {}
+   void createTexture(const char* image_path);
```
Comment (prince-chrismc) [inline NeverGreenGrove/NeverGreenGrove/src/Texture.h]: I think this should be private. you don't want to be able to change the loaded texture during the application.
make the ctor take a `const char *` and in the body call `CreateTexture(...)`

**Review by prince-chrismc — DISMISSED** [review]

File: NeverGreenGrove/NeverGreenGrove/src/NeverGreenGrove.cpp (hunk 1 above)
Comment (DJNGattuso) [reply NeverGreenGrove/NeverGreenGrove/src/NeverGreenGrove.cpp]: right!

File: NeverGreenGrove/NeverGreenGrove/src/Texture.h (hunk 2 above)
Comment (DJNGattuso) [reply NeverGreenGrove/NeverGreenGrove/src/Texture.h]: I can't say that I 100% understand what you mean...but I'll take a look at it

File: NeverGreenGrove/NeverGreenGrove/src/DrawableTree.cpp [hunk 3]
```diff
@@ -36,11 +36,15 @@ std::shared_ptr<TreeB::TreeObj> TreeB::TreeObj::s_Instance;
 
 TreeA::TreeObj::TreeObj()
 {
+	Texture temp("assets/wall.jpg");
```
Comment (prince-chrismc) [inline NeverGreenGrove/NeverGreenGrove/src/DrawableTree.cpp]: this is not the proper way of instantiating an object with a no default ctor... the correct was is to use an [initializer list](http://en.cppreference.com/w/cpp/language/initializer_list) this case would be like...
```c++
TreeA::TreeObj::TreeObj() : m_Texture("assets/wall.jpg") { /* body here */ }
```

Comment (DJNGattuso) [reply NeverGreenGrove/NeverGreenGrove/src/DrawableTree.cpp]: This, I didn't know. In my original commit, I wrote a comment asking if there's a better way to do this...now i know how!

**Pull request merged**